Sources/SWBTaskExecution/TaskActions/ClangCompileTaskAction.swift

```diff
@@ -483,29 +481,39 @@ public final class ClangCompileTaskAction: TaskAction, BuildValueValidatingTaskA
         isModular: Bool
     ) throws {
         let payload: DependencyValidationInfo.Payload
-        if let traceFilePath {
-            let traceFileContent = try fileSystem.read(traceFilePath)
+        if let traceFilePath,
+           let traceData = try parseTraceData(Data(fileSystem.read(traceFilePath))) {
 
-            let traceData: Array<TraceData>
-            // clang will emit an empty file instead of an empty array when there's nothing to trace
```

Comment:
Did you check that the empty file case works? I don't see an explicit check for that, but maybe it's handled by your parsing code.